pkg/plugins/golang/v3/scaffolds/internal/templates/readme.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import "sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+
+var _ machinery.Template = &Readme{}
+
+// Readme scaffolds a README.md file
+type Readme struct {
+	machinery.TemplateMixin
+	machinery.BoilerplateMixin
+	machinery.ProjectNameMixin
+}
+
+// SetTemplateDefaults implements file.Template
+func (f *Readme) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = "README.md"
+	}
+
+	f.TemplateBody = readmeFileTemplate
+
```

Comment:
So the string replacement should be done here. We need a new variable to pass the value to the template below.